src/lessons/02/Group.ts

```diff
@@ -0,0 +1,61 @@
+import { ClassBuilder, SortableArray } from '@/helpers'
+import { Status } from '@type/lesson02'
+import { Area, Student } from '@/lessons/02'
+
+type StaticFields = {
+  status: Status
+  area: Area
+  setStatus: (status: Status) => void
+  showPerformance: () => Student[]
+}
+type EditableFieldsList = {
+  student: Student
+}
+type IGroup = ClassBuilder<EditableFieldsList> & Readonly<StaticFields>
+
+export class Group implements IGroup {
+  private _students: SortableArray<Student> = new SortableArray<Student>()
+
+  constructor(
+    public directionName: string,
+    public levelName: string,
+    private _status: Status,
+    private readonly _area: Area,
+  ) {}
+
+  get area(): Area {
+    return this._area
+  }
+
+  get status(): Status {
+    return this._status
+  }
+
+  get students(): SortableArray<Student> {
+    return this._students
+  }
+
+  addStudent(student: Student): void {
+    this._students.push(student)
+  }
+
+  removeStudent(student: Student): void {
+    // any idea why have an error here?
+    // Property 'toSorted' is missing in type 'Student[]' but required in type 'SortableArray<Student>'
+    // but in real scenario after filtering we will get SortableArray after filtering, not plain Array
+
+    this._students = this._students.filter(
```

Comment:
It happens because filter method returns **new array**. This method doesn't care, that you created wrap around common Array, it returns Student[] in this case. And common array doesn't has method toSorted, that why you got this error. By the way, I think you fixed it in right way. Using `as` operator in this case is allowed.
![image](https://github.com/isprutfromua/hillel-ts/assets/39121701/dbe8c925-90c4-477f-ae37-c7b37ae0bda7)